File: 61-rotate-list/rotate-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* rotateRight(struct ListNode* head, int k) {
    struct ListNode* prev;
    struct ListNode* curr=head;
    int i,length=0,rk;
    if(head==NULL){
        return head;
    }
    if(head->next==NULL){
        return head;
    }
    while(curr!=NULL)
    {
        curr=curr->next;
        length++;
    }
    rk=k%length;
    curr=head;
    for(i=0;i<rk;i++)
    {
        prev=NULL;
        while(curr->next!=NULL)
        {
            prev=curr;
            curr=curr->next;
        }
        prev->next=NULL;
        curr->next=head;
        head=curr;
    }
    return head;
}

File: 61-rotate-list/rotate-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def rotateRight(self, head, k):
        if head is None:
            return head
        if head.next is None:
            return head
        curr=head
        length=0
        while curr is not None:
            curr=curr.next
            length+=1
        rk=k%length
        for i in range(rk):
            prev=None
            curr=head
            while curr.next is not None:
                prev=curr
                curr=curr.next        
            prev.next=None
            curr.next=head
            head=curr
        return head